packages/compose/index.js

```diff
@@ -127,37 +120,32 @@ function mergeComposable(dstDescriptor, srcComposable) {
  * @param {...(Composable)} [arguments] The list of composables.
  * @returns {Stamp} A new stamp (aka composable factory function)
  */
-module.exports = function compose() {
-  'use strict'; // to make sure `this` is not pointing to `global` or `window`
-  var descriptor = {};
-  var composables = [];
-  if (isComposable(this)) {
-    mergeComposable(descriptor, this);
-    composables.push(this);
-  }
+function compose(...args) {
+  const descriptor = {};
+  const composables = [];
+  const add = (value) => {
+    mergeComposable(descriptor, value);
+    composables.push(value);
+  };
 
-  for (var i = 0; i < arguments.length; i++) {
-    var arg = arguments[i];
-    if (isComposable(arg)) {
-      mergeComposable(descriptor, arg);
-      composables.push(arg);
-    }
-  }
+  if (isComposable(this)) add(this);
+
+  args.forEach((arg) => isComposable(arg) && add(arg));
```

Comment:
Can we please avoid `Array.prototype.forEach` altogether? I find `for-of` for readable. But most important - `for-of` is much more performant.

I even have a separate project where I measured things under multiple Node versions: https://github.com/koresar/bench-node

It clearly pointed me (at that time) that `.filter()` and `.map()` are much slower than `for` loop. The `for-of` is okay too.

Reply:
Now with `for (... of ...)` inside